packages/rebound-runtime/lib/hooks.js

```diff
@@ -38,64 +39,19 @@ function isComputedProperty(model, path){
   return _.isFunction(model.get(path, {raw: true}));
 }
 
-// Add a callback to a given context to trigger when its value at 'path' changes.
-function addObserver(context, path, lazyValue, morph) {
-  var length, res,
-      paths = $.splitPath(path);
-
-  if(!_.isObject(context) || !_.isString(path) || !_.isObject(lazyValue)){
-    console.error('Error adding observer for', context, path, lazyValue);
-    return;
-  }
-
-  // Get actual context if any @parent calls
-  while(paths[0] === '@parent'){
-    context = context.__parent__;
-    paths.shift();
-  }
-  path = paths.join('.');
-
-  // Ensure _observers exists and is an object
-  context.__observers = context.__observers || {};
-  // Ensure __obxervers[path] exists and is an array
-  context.__observers[path] = context.__observers[path] || [];
-
-  // Save the position this is being inserted into the observers array so we can garbage collect later
-  length = context.__observers[path].length;
-
-  // Add our callback
-  context.__observers[path].push(function() {
-    try{
-      return lazyValue.notify();
-    } catch(err) {
-      // If we run into an error running notify, that means we have a dead dependancy chain. Kill it.
-      console.log('KILLING OBSERVER', context.__observers, path, length);
-      console.log(err.stack);
-      lazyValue.destroy();
-      delete context.__observers[path][length];
-    }
-  });
-
-  res = context.get(lazyValue.path);
-
-  context.__observers[path][length].type = (res && res.isCollection) ? 'collection' : 'model';
-
-  return {context: context, path: path, index: length};
-}
-
 // Given an object (context) and a path, create a LazyValue object that returns the value of object at context and add it as an observer of the context.
 function streamProperty(context, path) {
 
   // Lazy value that returns the value of context.path
   var lazyValue = new LazyValue(function() {
     return context.get(path);
-  });
+  }, context);
```

Comment:
Context is passed to lazyvalue because you can now call .set() on any lazyvalue and have it update the root context with a new value. Used for input elements when binding their onchange events.